packages/fast-usdc/test/cli/lp-commands.test.ts

```diff
@@ -0,0 +1,111 @@
+import { Far, makeMarshal } from '@endo/marshal';
+import anyTest, { type TestFn } from 'ava';
+import { Command } from 'commander';
+import { flags } from '../../tools/cli-tools.js';
+import { mockStream } from '../../tools/mock-io.js';
+import { addLPCommands } from '../../src/cli/lp-commands.js';
+
+const makeTestContext = () => {
+  const program = new Command();
+  program.exitOverride();
+  const out = [] as string[];
+  const err = [] as string[];
+
+  const USDC = Far('usdcbrand');
+  const FastLP = Far('fastlpbrand');
+  const slotToVal = {
+    '0': USDC,
+    '1': FastLP,
+  };
+  const valToSlot = val => {
+    if (val === USDC) {
+      return '0';
+    }
+    if (val === FastLP) {
+      return '1';
+    }
+    return 'none';
+  };
+
+  const marshaller = makeMarshal(valToSlot, slot => slotToVal[slot]);
+  const now = () => 1234;
+
+  addLPCommands(program, {
+    vstorageKit: {
+      // @ts-expect-error fake brands
+      agoricNames: { brand: { FastLP, USDC } },
+      marshaller,
+    },
+    stdout: mockStream<typeof process.stdout>(out),
+    stderr: mockStream<typeof process.stderr>(err),
+    env: {},
+    now,
+  });
+
+  return { program, marshaller, out, err, USDC, FastLP, now };
+};
+
+const test = anyTest as TestFn<Awaited<ReturnType<typeof makeTestContext>>>;
+test.beforeEach(async t => (t.context = await makeTestContext()));
+
+test('fast-usdc lp deposit sub-command', async t => {
+  const { program, marshaller, out, err, USDC } = t.context;
+  const amount = 100.05;
+  const argv = [...`node fast-usdc deposit`.split(' '), ...flags({ amount })];
+  t.log(...argv);
+  await program.parseAsync(argv);
+
+  const action = marshaller.fromCapData(JSON.parse(out.join('')));
+  t.deepEqual(action, {
+    method: 'executeOffer',
+    offer: {
+      id: `lpDeposit-1234`,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: ['fastUsdc'],
+        callPipe: [['makeDepositInvitation', []]],
+      },
+      proposal: {
+        give: {
+          USDC: { brand: USDC, value: 100_050_000n },
+        },
+      },
+    },
+  });
+
+  t.is(
+    err.join(''),
+    'Now use `agoric wallet send ...` to sign and broadcast the offer.\n',
+  );
+});
+
+test('fast-usdc lp withdraw sub-command', async t => {
+  const { program, marshaller, out, err, FastLP, now } = t.context;
+  const amount = 100;
+  const argv = [...`node fast-usdc withdraw`.split(' '), ...flags({ amount })];
+  t.log(...argv);
+  await program.parseAsync(argv);
+
+  const action = marshaller.fromCapData(JSON.parse(out.join('')));
+  t.deepEqual(action, {
+    method: 'executeOffer',
+    offer: {
+      id: `lpWithdraw-1234`,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: ['fastUsdc'],
+        callPipe: [['makeWithdrawInvitation', []]],
+      },
+      proposal: {
+        give: {
+          PoolShare: { brand: FastLP, value: 100n },
```

Comment:
let's change that to `100_000_000`.

Reply:
Done in spirit, but I made the pool share worth > 1 to test the math/rounding logic of the offer proposal, so it's less than 100_000_000.